### PR TITLE
Migrate podman images to image library

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/image/image.go [hunk 1]
```diff
@@ -496,6 +500,32 @@ func (i *Image) History() ([]ociv1.History, []types.BlobInfo, error) {
 	return oci.History, img.LayerInfos(), nil
 }
 
+// Dangling returns a bool if the image is "dangling"
+func (i *Image) Dangling() bool {
+	dangling := false
```
Comment (mheon) [inline libpod/image/image.go]: Can simplify to `return len(i.Names()) == 0`

Comment (baude) [reply libpod/image/image.go]: fixed

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

File: libpod/runtime_img.go [hunk 2]
```diff
@@ -1306,51 +1217,40 @@ func ImageDangling() ImageResultFilter {
...
-				return true
-			}
+		labels, err := i.Labels()
+		if err != nil {
+			return false
+		}
+		if len(strings.TrimSpace(labels[key])) > 0 && len(strings.TrimSpace(value)) == 0 {
+			return true
+		}
+		if labels[key] == value {
```
Comment (rhatdan) [inline libpod/runtime_img.go]: Cleaner to do 

return labels[key] == value

File: libpod/runtime_img.go [hunk 3]
```diff
@@ -1306,51 +1217,40 @@ func ImageDangling() ImageResultFilter {
...
-			repository = name[0:li]
-			tag = name[li+1:]
-		}
-		if repository == i.Repository && len(strings.TrimSpace(tag)) == 0 {
-			return true
-		}
-		if repository == i.Repository && tag == i.Tag {
+func OutputImageFilter(userImage *image.Image) ImageResultFilter {
+	return func(i *image.Image) bool {
+		if userImage.ID() == i.ID() {
```
Comment (rhatdan) [inline libpod/runtime_img.go]: Ditto
return  userImage.ID() == i.ID()

File: cmd/podman/images.go [hunk 4]
```diff
@@ -253,35 +266,23 @@ func (i *imagesTemplateParams) HeaderMap() map[string]string {
...
-			if after.LocalName == "" {
 				return nil, errors.Errorf("unable to find image % in local stores", splitFilter[1])
 			}
-			img, err := r.GetImage(afterID)
+			filterFuncs = append(filterFuncs, libpod.ImageCreatedBefore(before.Created()))
+		case "after":
+			after, err := r.ImageRuntime().NewFromLocal(splitFilter[1])
 			if err != nil {
-				return nil, err
+				return nil, errors.Errorf("unable to find image % in local stores", splitFilter[1])
```
Comment (rhatdan) [inline cmd/podman/images.go]: errors.Wrapf(err, ...)

File: libpod/runtime_img.go (hunk 2 above)
Comment (baude) [reply libpod/runtime_img.go]: fixed

File: libpod/runtime_img.go (hunk 3 above)
Comment (baude) [reply libpod/runtime_img.go]: fixed

File: cmd/podman/images.go (hunk 4 above)
Comment (baude) [reply cmd/podman/images.go]: fixed

**Comment by vrothberg** [comment]
Tested locally, works like a charm.

**Comment by mheon** [comment]
LGTM, though I'd like to see the tests green before a merge

**Comment by baude** [comment]
bot, retest this please

**Comment by jlebon** [comment]
bot, retest this please

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**